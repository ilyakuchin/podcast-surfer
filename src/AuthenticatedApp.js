import React, { useContext } from "react";
import Podcast from "./components/Podcast/Podcast";
import EpisodePlayer from "./components/EpisodePlayer/EpisodePlayer";
import { Route, Link } from "react-router-dom";
import SearchPodcast from "./components/SearchPodcast/SearchPodcast";
import AuthContext from "./context/AuthContext";

export default function AuthenticatedApp() {
  const value = useContext(AuthContext);
  return (
    <div>
      {value.username}
      <Link to={"/"} onClick={value.logout}>
        Logout
      </Link>
      <Route path={"/"} component={SearchPodcast} exact />
      <Route path={"/podcast"} component={Podcast} />
      <Route path={"/episode-player"} component={EpisodePlayer} />
    </div>
  );
}
