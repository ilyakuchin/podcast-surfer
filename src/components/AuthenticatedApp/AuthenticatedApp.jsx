import React from 'react';
import { Route, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { logout } from '../../redux/actions/UserInfo/userInfo';
import ConnectedPodcast from '../Podcast/Podcast';
import ConnectedEpisodePlayer from '../EpisodePlayer/EpisodePlayer';
import ConnectedSearchPodcast from '../SearchPodcast/SearchPodcast';
import ConnectedSubscriptions from '../Subscriptions/Subscriptions';
import SubscriptionsEpisodeFeed from '../SubscriptionsEpisodeFeed/SubscriptionsEpisodeFeed';

const Container = styled.div`
  width: 100%;
  height: 100%;
`;

const UserProfile = styled.div`
  margin-right: 18px;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
`;

export function AuthenticatedApp({ username, logoutConnect }) {
  return (
    <Container>
      <UserProfile>
        {username}
        <div>
          <Link to='/subscriptions'>View subscriptions</Link>
        </div>
        <div>
          <Link to='/feed'>Feed</Link>
        </div>
        <div>
          <Link to='/' onClick={logoutConnect}>
            Logout
          </Link>
        </div>
      </UserProfile>
      <Route path='/' component={ConnectedSearchPodcast} exact />
      <Route path='/podcast' component={ConnectedPodcast} />
      <Route path='/episode-player' component={ConnectedEpisodePlayer} />
      <Route path='/subscriptions' component={ConnectedSubscriptions} />
      <Route path='/feed' component={SubscriptionsEpisodeFeed} />
    </Container>
  );
}

const mapStateToProps = state => {
  return { username: state.userInfo.username };
};

const dispatchToProps = dispatch => {
  return {
    logoutConnect: () => dispatch(logout())
  };
};

AuthenticatedApp.propTypes = {
  username: PropTypes.string.isRequired,
  logoutConnect: PropTypes.func.isRequired
};

const ConnectedAuthenticatedApp = connect(
  mapStateToProps,
  dispatchToProps
)(AuthenticatedApp);

export default ConnectedAuthenticatedApp;
