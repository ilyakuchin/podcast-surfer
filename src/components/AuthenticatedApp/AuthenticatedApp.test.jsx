import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { AuthenticatedApp } from './AuthenticatedApp';

configure({ adapter: new Adapter() });

describe('Test {AuthenticatedApp} component', () => {
  test('should render correctly', () => {
    expect(
      toJson(
        shallow(
          <AuthenticatedApp username='username' logoutConnect={() => {}} />
        )
      )
    ).toMatchSnapshot();
  });
});
