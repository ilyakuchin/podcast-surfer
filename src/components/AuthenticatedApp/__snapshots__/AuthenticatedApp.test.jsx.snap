// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test {AuthenticatedApp} component should render correctly 1`] = `
<div>
  username
  <Link
    to="/"
  >
    Logout
  </Link>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Podcast)",
        "type": [Function],
      }
    }
    path="/podcast"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(EpisodePlayer)",
        "type": [Function],
      }
    }
    path="/episode-player"
  />
</div>
`;
